facebook/ThreatExchange
# Add details on how to build docker based on local python-threatexchange

Summary
---------
In order to get docker to load the local version of `threatexchange` we first must copy the local version into the same directory as docker. We copy to a new directory called `local_threatexchange`.

In our dockerfile we now first check if this directory is a python module (does it have a `setup.py` file) and if so we copy the directory into our docker image. Lastly, we install the local package using pip. If we do this before we pip install the `requirements.txt` file the local version of `threatexhcange` is used rather than the published one

I added functions to the make file to do the above copy steps and also to delete the copied file after docker is made. Additionally edited the `make docker` command to create an empty directory because docker will fail if it tries to copy a directory that doesn't exist and there is no way to conditionally copy

Test Plan
---------
1. Edit the `fetcher.py` class to depend on some local changes to the `threatexchange` 
2. Add the following line to bottom of docker file to ensure that can be correctly executed 
```
RUN python3 ./hmalib/lambdas/fetcher.py
```
2. `$ make docker` fails why trying to run the `fetcher.py` file because it is pointed to the wrong version of `threatexchange` module
3. `$ make docker_local_te` and should succeed because it is pointed at the correct version of TE
